src/lib/events.ts

```diff
@@ -24,27 +49,49 @@ class EventListener<T extends unknown[]> {
 
 	/** @hidden */
 	public async call<T extends unknown[]>(...args: T): Promise<void> {
-		let lastCallReturn: unknown[] | undefined;
-
-		for (const handler of this.listeners) {
-			lastCallReturn = [handler(...(lastCallReturn ?? args))];
-		}
+		this._call(0, true, ...args);
 
 		if (this.yieldThreads.size() > 0) {
 			for (const thread of this.yieldThreads) coroutine.resume(thread);
 		}
 	}
+
+	private _call<T extends unknown[]>(iteration: number, conditionPassed?: boolean, ...args: T) {
+		if (iteration >= this.listeners.size()) return;
+
+		const [handlerOrCondition, action] = this.listeners[iteration];
+
+		switch (action) {
+			case EAction.COND:
+				this._call(iteration + 1, handlerOrCondition(), ...args);
+
+				break;
+
+			case EAction.DO:
+				if (conditionPassed) {
+					this._call(iteration + 1, conditionPassed, [...handlerOrCondition()]);
```

Comment:
Don't forget to try{}catch(e){}, and do whatever you want with the error, probably just log for now.